handlers/handlers.go

```diff
@@ -0,0 +1,197 @@
+package handlers
+
+import (
+	"database/sql"
+	"encoding/json"
+	"log"
+	"net/http"
+	"strconv"
+
+	"github.com/codingpop/simplerest/db"
+	"github.com/go-chi/chi"
+)
+
+// Handlers hold all the route handlers
+type Handlers struct {
+	db *db.DB
+}
+
+// New creates a new instance of Handlers
+func New(db *db.DB) *Handlers {
+	return &Handlers{
+		db: db,
+	}
+}
+
+// GetPosts fetches all posts
+func (h *Handlers) GetPosts(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+
+	posts, err := h.db.GetPosts(r.Context())
+	if err != nil {
+		log.Println(err)
+		http.Error(w, "internal server error", http.StatusInternalServerError)
+		return
+	}
+
+	if err := json.NewEncoder(w).Encode(map[string][]db.Post{"posts": posts}); err != nil {
+		http.Error(w, "failed to write response", http.StatusInternalServerError)
+	}
+}
+
+// GetPost fetches a particular post
+func (h *Handlers) GetPost(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+
+	id, err := strconv.Atoi(chi.URLParam(r, "id"))
+	if err != nil {
+		log.Println(err)
+
+		http.Error(w, "invalid id", http.StatusBadRequest)
+		return
+	}
+
+	post, err := h.db.GetPost(r.Context(), id)
+	if err != nil {
+		log.Println(err)
+
+		if err == sql.ErrNoRows {
```

Comment:
This will work, but the modern practice is to use the new `errors.Is` to recursively determine error equivalence in the presence of wrapped errors:
```suggestion
		if  errors.is(err, sql.ErrNoRows) {
```

See https://golang.org/pkg/errors/#pkg-overview.